src/main/java/bio/terra/workspace/app/configuration/ApiResourceConfig.java

```diff
@@ -11,4 +19,45 @@
   public void addResourceHandlers(ResourceHandlerRegistry registry) {
     registry.addResourceHandler("/**").addResourceLocations("classpath:/api/");
   }
+
+  @Autowired private SpanCustomizer spanCustomizer;
+
+  @Override
+  public void addInterceptors(InterceptorRegistry registry) {
+    registry.addInterceptor(
+        new HandlerInterceptor() {
+          @Override
+          public boolean preHandle(
+              HttpServletRequest httpRequest, HttpServletResponse httpResponse, Object handler)
+              throws Exception {
+
+            // get an mdc id from the request (if not found, create one), and pass it along in the response
+            String requestId = getMDCRequestId(httpRequest);
+            httpResponse.addHeader("X-Request-ID", requestId);
+
+            // add tags to Stackdriver traces
+            spanCustomizer.tag("requestId", requestId);
+
+            return true;
+          }
```

Comment:
@MatthewBemis  Moved MDC stuff from `Filter` into `preHandle`. I removed `MDC.put(...)`. Was that being used for something I was not aware of (I assumed it was just so that the `MdcResponseFilter` had access to it)

Reply:
Isn't the `MDC.put(...)` the critical piece that adds the id into the log statements? On this branch, does the id still appear in logs?

Reply:
Ahhh. Okay - my bad - I didn't take the time to learn about MDC fully and assumed what put() was for something else, which was incorrect.

Reply:
Adding the put() back in, the log statements do again show the mdc id. 
<img width="840" alt="Screen Shot 2020-07-23 at 10 40 59 AM" src="https://user-images.githubusercontent.com/17534065/88300682-c4c99d80-ccd1-11ea-8d74-bb1974331e57.png">